backend/tracim_backend/views/core_api/workspace_controller.py

```diff
@@ -924,6 +931,46 @@ def get_templates(
         )
         return api.get_templates(user=user, template_type=hapic_data.query["type"])
 
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_TRASH_AND_RESTORE_ENDPOINTS])
+    @hapic.handle_exception(EmptyLabelNotAllowed, HTTPStatus.BAD_REQUEST)
+    @check_right(can_delete_content_permanently)
+    @hapic.input_path(WorkspaceAndContentIdPathSchema())
+    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)
+    def delete_content_permanently(self, context, request: TracimRequest, hapic_data=None):
+        """
+        Delete a content. This route is for trusted users and administrators.
+        Note : a trusted user can only delete spaces on which he/she is space manager
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        session = request.dbsession
+        path_data = hapic_data.path
+
+        api = ContentApi(
+            config=app_config, session=session, current_user=request.current_user, show_deleted=True
+        )
+
+        content = api.get_one(path_data.content_id, content_type=ContentTypeSlug.ANY.value)
+
+        if is_administrator or content.is_deleted:
+            if session.bind.dialect.name == "sqlite":
+                listen(Engine, "connect", enable_sqlite_foreign_keys)
```

Comment:
Are we forced to use  `sqlalchemy.event.listen` ? It looks like a vulnerable way because event mechanisms are difficult to maintain. I don't know the entire context on this work, but I think it could be a context manager which disable/(enable ? cf. https://github.com/tracim/tracim/pull/6377/files#r1504005867) foreign keys on given `session` object.

Reply:
done